### DISCUSSION TITLE
Dropping Python 3.6 from CI

### DISCUSSION THREAD

**Discussion by auvipy** [encode/django-rest-framework#8814] [Polls]
Python 3.6 has been EOL for long time. the latest release also support it. however many dependencies started to dropping support for python 3.6 so we are having issues in CI. I want community input to gauge the interest and usage of python 3.6 in main branch. In my humble opinion, dropping it from CI only would be easier for the CI with no code change. the next version 3.15 will be python 3.7+ only (which will be EOL as well by June 2023). users with need of python 3.6 support can still stick with 3.14.

**Comment by onegreyonewhite** [comment]
I think @tomchristie came up with the most adequate python version support plan to follow in everything.

**Reply by auvipy** [reply]
that was for 3.14 release. now many dependencies are dropping py 3.6 support. that is why it was raised.

**Reply by auvipy** [reply]
also django 4.2a1 is due in january 2023

**Reply by onegreyonewhite** [reply]
> also django 4.2a1 is due in january 2023

Do you create any app on alpha version of lib?) Stable release is due April.

**Reply by onegreyonewhite** [reply]
> that was for 3.14 release. now many dependencies are dropping py 3.6 support. that is why it was raised. 

That was for project as I can see. If it plan to release 3.15 at April with Django 4.2 all my hands up for drop python 3.6.

**Reply by auvipy** [reply]
you also not use from development branch right? the proposed change will not effect until we release 3.15 with django 4.2 support. at that time another django release will be dropped/or announced unsupported. DRF usually start testing with alpha versions and keep fixing issues raised by the new alpha/beta/rc releases in development branch.

**Comment by peterthomassen** [comment]
I'd like to reraise this. I think our current DRF version should run on the same Python versions as the current Django release. [According to their docs](https://docs.djangoproject.com/en/5.0/faq/install/#faq-python-version-support), this is currently Python 3.10+.

Rationale: People running old Django with old Python can also run old DRF. I'm not sure why we should maintain old Python compatibility when even Django doesn't anymore.

If folks don't share this view, I think we can at least drop support for now unsupported Django releases (and associated Python versions). The oldest supported Django release series is 4.2 LTS, which supports Python 3.8+. This means that we can drop Python 3.6 and 3.7.

This would allow us to upgrade pygments, [which is currently pinned](https://github.com/encode/django-rest-framework/pull/9393/commits/146887354ecf7f80267e2a6de0da6213580a3b83) because of Python 3.6. Doing this would reduce maintenance overhead (keeping track of pins, dealing with Dependabot alerts etc.), and would overall increase hygiene.

@tomchristie

**Reply by lovelydinosaur** [reply]
> The oldest supported Django release series is 4.2 LTS, which supports Python 3.8+. This means that we can drop Python 3.6 and 3.7.

👍🏼

**Reply by onegreyonewhite** [reply]
Yep! This is a great idea👍🏼

**Reply by auvipy** [reply]
I want to share some of my views or just like to have some idea. should we keep at least 2 LTS supported at a time unless it is too hard to maintain? as DRF is now mostly in maintenance mode.

**Reply by cclauss** [reply]
@auvipy I am not a fan of actively enabling the use of unsupported software to run critical infrastructure.  If users of this software do not want to upgrade their tools, they can continue using old DRF versions like they continue to use Python 3.6.  Slowing and complicating the improvement and testing of DFR to cater to this risk-taking audience seems suboptimal.

**Reply by peterthomassen** [reply]
PR: https://github.com/encode/django-rest-framework/pull/9393
